### DISCUSSION TITLE
Allow aligning left/right Y axis without aligning ticks

### DISCUSSION THREAD

**Discussion by andig** [grafana/grafana#38371] [Dashboards, Panels & Visualizations]
**What would you like to be added**:

Aligning Y axes was introduced with https://github.com/grafana/grafana/pull/10994. This PR apparently also aligns ticks by a certain factor. I'd propose to not align ticks by a fixed factor and instead let each axis scale individually.

**Why is this needed**:

Consider this panel:

![0A12FBDE-0B46-4B0A-BBF2-869ED6E9F550_1_105_c](https://user-images.githubusercontent.com/184815/78880541-da35f700-7a55-11ea-92fb-1b4a19575cb0.jpeg)

Once both axes are aligned, the left axis looses resolution since apparently both axes are tied together by a factor of 100. If the left axis as allowed to scale to 600 while the right scales to 100 the chart are would be more efficiently used.

**Comment by andig** [comment]
Here's another screenshot that shows that the current mode for aligning the axes is sometimes not too helpful:

<img width="1151" alt="Screenshot 2020-04-11 at 13 04 50" src="https://user-images.githubusercontent.com/184815/79042098-24a1aa00-7bf5-11ea-93a1-711bafd1e0cc.png">

Going forward it might make sense to have two different alignment modes: `Zero` and `Zero and Ticks`. The latter is the current behaviour.

**Comment by Pwuts** [comment]
I think related to this is the issue that, even using the "align Y-axes" option, the axes are aligned to ticks instead of the exact level specified. So instead of a ratio of 1.78, I get a ratio of e.g. 1.7 or 1.8 between the axes, causing the lines to still be misaligned. When I set a fixed Y-max on both axes, this is not the case.

Y_min1 = -1; Y_min2 = -1.78; align y-axes: true, level = 0 --> aligned to ticks == misaligned
Y_min1 = 0; Y_min2 = 0; align y-axes: false; Y_max1: 100; Y_max2: 178 --> aligned properly

My use case is quite simple: I just want the Y-axis to have a fixed ratio between them, without setting a hard maximum. For example, the left Y-axis would show the number of kWh's consumed, and the right Y-axis the exact price of this consumed energy. The price per kWh is fixed, and so should the ratio between Y1 and Y2 be.

Should I create a new issue for this or is it basically the same issue as this one?

**Reply by Pwuts** [reply]
Years later and this is still something I could really use. It could be implemented by specifying a second alignment level in "Align Y-axes", allowing to set a ratio between them without specifying the maximum for both axes.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by leeoniya** [comment]
> This PR apparently also aligns ticks by a certain factor. I'd propose to not align ticks by a fixed factor and instead let each axis scale individually.

we added a Centered Zero option in https://github.com/grafana/grafana/pull/52555 which does not require setting explicit limits and does a good job of covering the most common need for axis alignment. there are some other improvements to be made in https://github.com/grafana/grafana/issues/54343, but they're somewhat technically involved.

cc @Pwuts

**Reply by Pwuts** [reply]
I'm not sure what you mean w.r.t. my proposal for fixed ratio scaling on the Y-axes. Is it technically involved to implement?

**Reply by leeoniya** [reply]
currently, all scales and series that belong on them are fully independent of each other. making them interdependent is not trivial and carries large performance degradation risks.

> allowing to set a ratio between them

you then need to at least have a way to set one of them to be a "master" axis. what happens when their scales are incompatible (log and linear)? what happens when you also have soft and/or hard limits set? what if centered zero is toggled? what about decimals: 0? what if the master axis is toggled off as a result of all its series being toggled off? there are dozens of questions that dont have obvious answers. you can't just implement new features in isolation from everything else that exists and has to continue to interoperate logically.

**Reply by Pwuts** [reply]
Thanks for clarifying!

**Reply by Pwuts** [reply]
> currently, all scales and series that belong on them are fully independent of each other

With that in mind, it might be easier to implement an option to add extra scales, instead of fixed-ratio double axes.

Ssomething like this:
![image](https://user-images.githubusercontent.com/12185583/212720422-a89806a1-b9a7-4701-82b9-97ecc144a278.png)

Here the "normal" axis (black) is *always* the master, and extra scales (red/blue here) can be added to the same axis with
* a ratio + zero point or a formula giving its value with respect to the master axis
* a unit
* whether to show the value of a series on a particular scale in the tooltip

This way, the axes can remain independent but linearly related values can still be displayed/visualized together in one graph. Does this sound doable?

**Reply by leeoniya** [reply]
if these are display-only reference scales, then this is easier, and i have a demo of this using the underlying plot library [1]. i'm not sure this is a common enough requirement to add a lot more options to the TimeSeries panel, it's already bursting with options!

[1] https://leeoniya.github.io/uPlot/demos/dependent-scale.html

**Comment by natellium** [comment]
Hey there! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

I will be closing this discussion, but if you’re still interested in this request being implemented please don’t hesitate to open a [feature request](https://github.com/grafana/grafana/issues/new?assignees=&labels=&projects=&template=1-feature_requests.md). Sorry for the inconvenience, and thanks for your contributions!

cc @grafana/dataviz-squad
